flang/lib/Lower/Allocatable.cpp

```diff
@@ -171,7 +200,7 @@ class AllocateStmtHelper {
     const auto &allocObj = std::get<Fortran::parser::AllocateObject>(alloc.t);
     const auto &symbol = unwrapSymbol(allocObj);
     assert(symbol.GetType());
-    return Allocation{alloc, symbol, *symbol.GetType()};
+    return Allocation{alloc, *symbol.GetType()};
```

Comment:
Is the name of the symbol of importance to debugging or other metadata (aka attributes)?

Reply:
Does dwarf has special metadata around allocate ?
Otherwise, I think the name is not important so far. The runtime does not use it to build the error messages.

Reply:
The name is now used when generating `fir.alloc_mem`.